weather_display.py

```diff
@@ -0,0 +1,130 @@
+import calendar
+
+
+class WeatherDataPresenter:
+    @staticmethod
+    def show_year_extremes(weather_extremes):
+        """
+        Print yearly extremes for maximum temperature, minimum temperature, and
+        mean humidity.
+
+        Args:
+        - weather_extremes (dict): Dictionary containing the extremes.
+          Keys: "Highest", "Lowest", "Humidity".
+          Values: Tuples (extreme_value, extreme_day) for each category.
+        """
+        (highest_temperature,
+         highest_temperature_day) = weather_extremes["Highest"]
+
+        (lowest_temperature,
+         lowest_temperature_day) = weather_extremes["Lowest"]
+
+        (max_humidity_value,
+         most_humid_day) = weather_extremes["Humidity"]
+
+        highest_temperature_day = highest_temperature_day.split("-")
+        lowest_temperature_day = lowest_temperature_day.split("-")
+        most_humid_day = most_humid_day.split("-")
+
+        highest_temperature_month = highest_temperature_day[1]
+        lowest_temperature_month = lowest_temperature_day[1]
+        most_humid_month = most_humid_day[1]
+
+        highest_temperature_day = highest_temperature_day[2]
+        lowest_temperature_day = lowest_temperature_day[2]
+        most_humid_day = most_humid_day[2]
+
+        print(
+            f"Highest: {highest_temperature}C on "
+            f"{calendar.month_abbr[int(highest_temperature_month)]} "
+            f"{highest_temperature_day}"
+        )
+        print(
+            f"Lowest: {lowest_temperature}C on "
+            f"{calendar.month_abbr[int(lowest_temperature_month)]} "
+            f"{lowest_temperature_day}"
+        )
+        print(
+            f"Humidity: {max_humidity_value}% on "
+            f"{calendar.month_abbr[int(most_humid_month)]} "
+            f"{most_humid_day}"
+        )
+
+    @staticmethod
+    def show_month_averages(month_weather_averages):
```

Comment:
This method is displaying or printing. Show is a vague word which is not revealing the right intent.